### PR TITLE
dec-decode: init at unstable-2022-12-24

### PR DESCRIPTION
**Pull request by hughobrien**
###### Description of changes

Add [dec-decode](https://github.com/sammiq/dec-decode) which assists with decoding ISOs from older nintendo consoles.

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/tools/dec-decode/default.nix [hunk 1]
```diff
@@ -0,0 +1,26 @@
+{ lib
+, buildGoModule
+, fetchFromGitHub
+}:
+
+buildGoModule rec {
+  pname = "dec-decode";
+  version = "2023-04-03-git";
```
Comment (fabaff) [inline pkgs/development/tools/dec-decode/default.nix:8]:
```suggestion
  version = "unstable-2023-04-03";
```

File: pkgs/development/tools/dec-decode/default.nix [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+  src = fetchFromGitHub {
+    owner = "sammiq";
+    repo = pname;
+    rev = "6ee103c";
+    hash = "sha256-vOYKVl00oaXwp77pRLqUPnXQT5QuJfqnGGkQVBMq5W0=";
+  };
+
+  vendorSha256 = "sha256-zGWRzw1KUmifIsTudlgoKCR3+K0FLehHRSB3lNX+OWY=";
```
Comment (fabaff) [inline pkgs/development/tools/dec-decode/default.nix:17]:
```suggestion
  vendorHash = "sha256-zGWRzw1KUmifIsTudlgoKCR3+K0FLehHRSB3lNX+OWY=";
```

File: pkgs/development/tools/dec-decode/default.nix [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+  meta = with lib; {
+    description = "nintendo wii iso.dec decoder";
```
Comment (fabaff) [inline pkgs/development/tools/dec-decode/default.nix:20]:
```suggestion
    description = "Nintendo Wii iso.dec decoder";
```

File: pkgs/development/tools/dec-decode/default.nix (hunk 1 above)
Comment (hughobrien) [reply pkgs/development/tools/dec-decode/default.nix:8]: accepted

File: pkgs/development/tools/dec-decode/default.nix (hunk 2 above)
Comment (hughobrien) [reply pkgs/development/tools/dec-decode/default.nix:17]: accepted

File: pkgs/development/tools/dec-decode/default.nix [hunk 4]
```diff
@@ -0,0 +1,26 @@
+{ lib
+, buildGoModule
+, fetchFromGitHub
+}:
+
+buildGoModule rec {
+  pname = "dec-decode";
+  version = "unstable-2023-04-03";
```
Comment (wegank) [inline pkgs/development/tools/dec-decode/default.nix:8]:
```suggestion
  version = "unstable-2022-12-24";
```

Comment (hughobrien) [reply pkgs/development/tools/dec-decode/default.nix]: included in forced push. thank you!

**Pull request merged**